code/modules/shuttles/marine_ferry.dm

```diff
@@ -590,7 +590,76 @@
 	if(SSticker.mode)
 		SSticker.mode.is_in_endgame = TRUE
 		SSticker.mode.force_end_at = world.time + 15000 // 25 mins
+//////
+////// Le funny immersive crash sequence
+////// independent of main crash proc so sleep()s don't matter here. Try keep it under 15s as we only have 45s before crash and we run on BYOND time. We wouldn't want
+////// the ds still getting hit during,and after the crash.
 
+////// No combat is really occuring during this point of the round so it should give us more performance overhead to give a nice fireworks display (:
+/datum/shuttle/ferry/marine/proc/AA_sequence(turfs_int)
+	set waitfor = 0
+	var/sound_to_play = pick(SOUND_DISTANT_AA_FIRE)
+	var/sound_to_play2 = pick(SOUND_DISTANT_AA_FIRE)
+	//init the funny spark system
+	var/datum/effect_system/spark_spread/s = new
+	var/datum/effect_system/spark_spread/s2 = new
+	var/datum/effect_system/spark_spread/s3 = new
+	var/turf/T
+	var/mob/M
+	var/cause = create_cause_data("IX-50 MGAD")
+	// Used to define shuttle area during crash
+	var/area/shuttle_area
+	var/ship_zlevels = SSmapping.levels_by_trait(ZTRAIT_MARINE_MAIN_SHIP)
+	// define all shuttle turfs as shuttle_area. This includes some deadspace outside of DS, in which some effects may randomly spawn there. Consider that variety, if you will.
+	if(!shuttle_area)
+		for(T as anything in turfs_int)
+			if(T?.loc)
+				shuttle_area = T.loc
+				break
+	marine_announcement("A hostile aircraft has been detected on course for the [true_crash_target_section]. IX-50 MGAD & ASAT missiles are locked onto target trajectory. Firing for effect.", "ARES")
+	playsound_z(ship_zlevels, 'sound/effects/alert.ogg', 75, echo = list(-500, -2000, 1000, 500, 0, 0.8, 5, 5), y_s_offset = 10)
+	for(M in shuttle_area)
+		if(isXeno(M))
+			to_chat(M, SPAN_HIGHDANGER("Alarms blare throughout the bird! The fleshy hosts are definately trying something!"))
+		else
+			to_chat(M, SPAN_HIGHDANGER("Lock-on alarms and missile warnings blare from the cockpit! Oh crap!"))
```

Comment:
This micro optimization is just not worth it. Just define M here instead of defining it early on